### ISSUE TITLE
MAC CPU Usage

### ISSUE THREAD

**Issue by tehdef** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#2320]
[debug_info.zip](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/files/844836/debug_info.zip)

**OS:** MacOS 10.12.4

**Issue Descriptions:** Activity Monitor is registering 2 instances of GPMDP Helper and 1 instance of GPMDP. Total CPU usage is 10-15% for EACH process and is higher than any other app in usage. Combined is 40% of my CPU just to play 1 song... Additionally, Memory shows 3 instances of the helper ranging from 50-200mb of RAM, and the core app using 113. 

This has to be addressed. System resources are important. The footprint for this should be minimal. 

**Steps to Reproduce:** Load. View activity Monitor. Realize your CPU is tied up and your Macbook is running at 120 degrees Fahrenheit because of it.

**Comment by MarshallOfSound** [comment]
As discussed in numerous other issues there is minimal impact I can have on this resource usage.  Admittedly your described usage is well outside of normal bounds (my usage hovers between 1 and 8% cpu) it is probably due to the size of your music library or the age of your mac or something else out of my control.

GPMDP's code is heavily optimized.  Lot's of the resource usage comes from GPM itself which is outside of my control

**Comment by tehdef** [comment]
While I understand why this was closed, maybe I just need more help understanding the why so many Helpers are active? 

What are they doing? Is it something that GPMDP can control? Is the only thing that GPMDP controls is the actual GPMDP app? Or are the helpers directly connected? I turned off the JSON API and now the GPMDP app runs between 4.4-10% max. It's the helper that is eating 20% or so of the CPU usage. 

For information purposes, I have a mid 2012 Core i7 Macbook Pro. 

Is this simply a closed issue, or is there more I can do to help better understand the app and how it uses my resources? 

Thanks Sam.
